### PR TITLE
Support adding custom commands at compile time

### LINKED ISSUE #162: Support custom commands defined in Redis modules
I'm trying to use redis modules with hiredis-cluster. In previous version of hiredis-cluster, it is possible to add commands to command.h in order to use the redis commands like FT.CREATE which comes from redisearch. But in the last version, there is a script which is named as gencommands.py to produce cmddef.h. As far as I know, I need to add commands to this file to produce QUERY to send redis server. But I couldn't have make it for RediSearch and RedisJSON. I'm giving the json which is located in redisjson and redisearch as an argument my I'm taking an error. How can I use redisearch and redisjson with hiredis-cluster?

### PR DESCRIPTION
**Pull request by zuiderkwast**
Extend gencommands.py with more flexible command handling:

* Don't require the JSON filename to match the command name.
* Don't require command name to be in uppercase in the JSON file.
* Allow non-alphanumeric chars in command name.
* Accept multiple JSON files on the command line.
* For convenience, accept files on the format of cmddef.h as input.
* Handle JSON files without key specs, with a fallback to using the "arguments"
  key to find the position of the first key.
* Handle JSON files where subcommands are defined as a simple command with a
  space in the command name (instead of as a nested JSON structure).

Additionally, correct uppercase ordering of commands where the ordering would
be ambiguous if strncasecmp() would have been used (e.g. EVAL_RO vs EVALSHA).
This fixes possible failures to find such commands in the lookup table resulting
in "unknown command" errors.

Fixes #162.

### CODE REVIEW

**Review by bjosv — COMMENTED** [review]
Fantastic.
`alloca` was for me a forgotten gem, but seems sweet in this case.

File: gencommands.py [hunk 1]
```diff
@@ -7,12 +7,16 @@
 # describing the commands. This is done manually when commands have been added
 # to Redis.
 #
-# Usage: ./gencommands.py path/to/redis > cmddef.h
+# Usage: ./gencommands.py path/to/redis/src/commands/*.json > cmddef.h
+#
+# Additional JSON files can be added to define custom commands. For convenience,
```
[the remark this answers is not in the document]
Comment (zuiderkwast) [reply gencommands.py:12]: Good idea. Any idea where in the README would be suitable? Under build details?

[the remark this answers is not in the document]
Comment (zuiderkwast) [reply gencommands.py:12]: Added. Please have a look.

**Comment by zuiderkwast** [comment]
@selcukaltinay FYI

File: gencommands.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjosv) [reply gencommands.py:12]: Looks good

**Review by bjosv — APPROVED** [review]

**Pull request merged**